### PR TITLE
[Mount] Add ChunkGroup seeking tests and fix boundary handling

### LINKED ISSUE #11221: [Mount] Add test coverage for ChunkGroup data and hole seeking
### Problem Description

`ChunkGroup.SearchChunks` is a core part of Mount/FUSE file seeking. It implements `SEEK_DATA` and `SEEK_HOLE` to locate data ranges and holes within a file.

The existing tests mainly cover empty `ChunkGroup` instances and context cancellation, but do not fully validate data-range seeking. The implementation also explicitly marks this area for additional tests: [weed/filer/filechunk_group.go:265](https://github.com/seaweedfs/seaweedfs/blob/master/weed/filer/filechunk_group.go#L265).

Image

Sparse files, data/hole transitions, section boundaries, and end-of-file behavior are not sufficiently covered. As a result, future changes could introduce incorrect seek offsets without being detected.

This task will first add tests to define the expected behavior of `SearchChunks`. If the tests reveal implementation issues, they will be fixed within the same scope.

### Scope

- Add table-driven tests for `SearchChunks`;
- Cover sparse files, data/hole transitions, section boundaries, empty files, and end-of-file behavior;
- Fix only issues revealed by the new tests;
- Do not modify the storage format or public APIs.

### Impact Radius

This only affects Mount/FUSE file seeking behavior. Normal file reads and writes are not affected.

### Tests

```bash
go test ./weed/filer/... ./weed/mount/...
```

### PR DESCRIPTION
**Pull request by ssshr-66**
fix [#11221](https://github.com/seaweedfs/seaweedfs/issues/11221)

# What problem are we solving?

`ChunkGroup.SearchChunks` is used by Mount/FUSE to implement `SEEK_DATA` and `SEEK_HOLE`. Its boundary behavior was not sufficiently tested, and some sparse-file and Section boundary cases could return incorrect offsets.

# How are we solving the problem?

- Add table-driven tests for data and hole seeking.
- Cover sparse files, data/hole transitions, Section boundaries, missing Sections, empty files, and EOF behavior.
- Fix incorrect offsets returned before a data range.
- Correct `SEEK_DATA` and `SEEK_HOLE` handling across Section boundaries.
- Keep the changes limited to ChunkGroup seeking behavior and tests.
- Do not modify storage formats or public APIs.

# How is the PR tested?

- `go test ./weed/filer/... ./weed/mount/...`
- Added regression tests covering the cases described above.
image

# Checks

- [x] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.
- [x] All AI code review comments have been addressed. No more comments to fix if reviewed again. Reviewer may request additional gemini and copilot reviews.

# Checks for AI generated PRs

- [x] I have reviewed every line of code.
- [x] The PR is kept as minimum as possible. Large PRs would not be accepted.

---

<a href="https://app.devin.ai/review/seaweedfs/seaweedfs/pull/11223" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-devin-review-dark.svg?v=3">
    Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved file data and hole detection across section boundaries.
  * Correctly handles partial sections, missing sections, gaps, empty files, and end-of-file ranges.
  * Data searches now return the next available data interval when the requested position falls within a gap.
  * `SEEK_DATA` now correctly reports `ENXIO` when no data exists at or after the requested position.

* **Tests**
  * Expanded coverage for data and hole seeking across boundary, gap, missing-section, EOF, and empty-file scenarios.

### CODE REVIEW

File: weed/filer/filechunk_group.go [hunk 1]
```diff
@@ -273,32 +272,37 @@ func (group *ChunkGroup) SearchChunks(ctx context.Context, offset, fileSize int6
...
+		}
+		holeStart := section.NextStopOffset(ctx, group, sectionStart, fileSize)
+		if holeStart < sectionStop {
 			return true, holeStart
 		}
-		return true, fileSize
 	}
+
+	if whence == SEEK_DATA {
+		return false, 0
```
[the remark this answers is not in the document]
Comment (ssshr-66) [reply weed/filer/filechunk_group.go:305]: Added a regression test for SEEK_DATA when no data exists at or after the requested offset, and fixed the Mount layer to return ENXIO instead of the original offset. The test passes.

File: weed/filer/filechunk_group.go [hunk 2]
```diff
@@ -273,32 +272,37 @@ func (group *ChunkGroup) SearchChunks(ctx context.Context, offset, fileSize int6
 func (group *ChunkGroup) doSearchChunks(ctx context.Context, offset, fileSize int64, whence uint32) (found bool, out int64) {
 
 	sectionIndex, maxSectionIndex := SectionIndex(offset/SectionSize), SectionIndex(fileSize/SectionSize)
-	if whence == SEEK_DATA {
-		for si := sectionIndex; si < maxSectionIndex+1; si++ {
-			section, foundSection := group.sections[si]
+	for si := sectionIndex; si <= maxSectionIndex; si++ {
+		sectionStart := max(offset, int64(si)*SectionSize)
+		sectionStop := min(fileSize, int64(si+1)*SectionSize)
```
[the remark this answers is not in the document]
Comment (ssshr-66) [reply weed/filer/filechunk_group.go]: Added boundary tests for the final section at math.MaxInt64 and fixed section bounds to use the actual file size. The tests pass.
